include/Utilities.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <string>
+#include <map>
+
+using std::cout;
+using std::cin;
+using std::endl;
+using std::string;
+using std::unique_ptr;
+using std::vector;
+using std::make_unique;
+using std::move;
+
+class Piece;
+
+using board_vec = vector<vector<unique_ptr<Piece>>>;
+using board_vec_const = const vector<vector<unique_ptr<Piece>>>;
```

Comment:
This is considered bad practice, try not to name `const` or `const ref` types if you don't have to. It's always better to pass the argument as simply `const board_vec` or `const board_vec &`